pkg/handlers/config/config.go

```diff
@@ -1,69 +1,213 @@
+func (cmd *GetContextsCmd) Run(c *cc.CommonCtx) error {
+	configFile, err := config.GetConfigFile()
+	if err != nil {
+		return err
+	}
+
+	cfg, err := config.GetConfigFromFile(configFile)
+	if err != nil {
+		return err
+	}
+
+	return jsonx.OutputJSON(c.UI.Output(), cfg.Context)
 }
 
-type GetTenantCmd struct{}
+type GetActiveContextCmd struct{}
 
-func (cmd *GetTenantCmd) Run(c *cc.CommonCtx) error {
-	client, err := c.TenantClient()
+func (cmd *GetActiveContextCmd) Run(c *cc.CommonCtx) error {
+	configFile, err := config.GetConfigFile()
 	if err != nil {
 		return err
 	}
 
-	req := &account.GetAccountRequest{}
+	cfg, err := config.GetConfigFromFile(configFile)
+	if err != nil {
+		return err
+	}
 
-	resp, err := client.Account.GetAccount(c.Context, req)
+	for _, ctxs := range cfg.Context.Contexts {
+		if cfg.Context.ActiveContext == ctxs.Name {
+			return jsonx.OutputJSON(c.UI.Output(), ctxs)
+		}
+	}
+
+	return nil
+}
+
+type DeleteContextCmd struct {
+	ContextName string `arg:"" name:"context_name" required:"" help:"context name"`
+}
+
+func (cmd *DeleteContextCmd) Run(c *cc.CommonCtx) error {
+	configFile, err := config.GetConfigFile()
 	if err != nil {
-		return errors.Wrapf(err, "get account")
+		return err
 	}
 
-	type tenant struct {
-		ID      string `json:"id"`
-		Name    string `json:"name"`
-		Current bool   `json:"current"`
-		Default bool   `json:"default"`
+	cfg, err := config.GetConfigFromFile(configFile)
+	if err != nil {
+		return err
 	}
 
-	tenants := make([]*tenant, len(resp.Result.Tenants))
+	if cfg.Context.ActiveContext == cmd.ContextName {
+		return errors.Errorf("the context in use cannot be deleted")
+	}
 
-	for i, t := range resp.Result.Tenants {
-		isCurrent := (t.Id == c.TenantID())
-		isDefault := (t.Id == resp.Result.DefaultTenant)
-		tt := tenant{
-			ID:      t.Id,
-			Name:    t.Name,
-			Current: isCurrent,
-			Default: isDefault,
+	for index, ctxs := range cfg.Context.Contexts {
+		if ctxs.Name == cmd.ContextName {
+			cfg.Context.Contexts = append(cfg.Context.Contexts[:index], cfg.Context.Contexts[index+1:]...)
+			break
 		}
-		tenants[i] = &tt
 	}
 
-	return jsonx.OutputJSON(c.UI.Output(), tenants)
+	data, err := json.Marshal(cfg)
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(configFile, data, 0600)
 }
 
-type SetTenantCmd struct {
-	ID      string `arg:"" required:"" name:"tenant-id" help:"tenant id"`
-	Default bool   `name:"default" help:"set default tenant for user"`
+type SetContextCmd struct {
+	Context  string `arg:""  type:"existingfile" name:"context" optional:"" help:"file path to context or '-' to read from stdin"`
+	Template bool   `name:"template" help:"prints a context template on stdout"`
+	Force    bool   `name:"force" help:"if a context wit the same name exists, forces overwrite"`
 }
 
-func (cmd *SetTenantCmd) Run(c *cc.CommonCtx) error {
+func (cmd *SetContextCmd) Run(c *cc.CommonCtx) error {
+	if cmd.Template {
+		return printContext(c.UI)
+	}
+
+	if cmd.Context == "" {
```

Comment:
If `Context` is a required parameter, why mark it as `optional:""` in the struct?

Reply:
in order for the `--template` flag to work

Reply:
Then perhaps printing a template should be it own command instead of a flag on set-context?